invenio_records_rest/views.py

```diff
@@ -433,7 +433,13 @@ def get(self, **kwargs):
 
         # Arguments that must be added in prev/next links
         urlkwargs = dict()
-        search = self.search_class().params(version=True)
+        self.search_index = current_app.config[
```

Comment:
This makes more sense now, the only problem I can think of is that someone tries to modify the search_index in the config.py after partial has already been called and the changes are not registered. If this is not supported then these changes aren't needed and I can close this.